### PR TITLE
Adds #324

### PR DESCRIPTION
**Pull request by boonedev**
If approved this PR:
- Adds email opt-in panel 
- Fixes misc phpcs issues

### CODE REVIEW

**Comment by SteveJonesDev** [comment]
@boonedev, is the form supposed to disappear after being filled out? Refreshing the page doesn't bring it back either.

**Comment by SteveJonesDev** [comment]
Looks like it's per user.

**Comment by boonedev** [comment]
Sounds like a bug. It should:
1) show the form
2) if user email submit is good, the form should thank them, then hide the form and not display again.

Which browser?

**Comment by boonedev** [comment]
Also, what do you think about pre-populating the form with their email as a convenience?

**Comment by SteveJonesDev** [comment]
Using the Brave browser.

1. The form is shown
2. I enter an email address
3. It shows a thank you message
4. Then after about a second, the form is dynamically removed from the dom

**Comment by boonedev** [comment]
So thank-you/hide happens before the submit?

**Comment by SteveJonesDev** [comment]
Sorry, no I missed a step. I did click the submit button. I don't think the form should be dynamically removed from the page. I'm not sure if you're doing that or Zoho.

File: includes/classes/class-welcome-page.php [hunk 1]
```diff
@@ -205,4 +205,144 @@ public static function render_summary() {
...
+							<span style="display: none" id="dt_CONTACT_EMAIL">1,true,6,Contact Email,2</span>
+							</form>
+						</div>
+						<div class="opt-in-show" style="display: none;">
+							<h2>' . __( 'Get Notified of Upcoming Events', 'accessibility-checker' ) . '</h2>
+							<div>' . __( 'Thank-you for signing up!', 'accessibility-checker' ) . '</div>
+						</div>
+					</div>
+				</div>
+				<img src="https://zmp-glf.maillist-manage.com/images/spacer.gif" id="refImage" onload="referenceSetter(this)" style="display:none;">
```
Comment (SteveJonesDev) [inline includes/classes/class-welcome-page.php:278]: Accessibility: A layout spacer image (which should have null/empty alternative text) does not have an alt attribute.

**Comment by boonedev** [comment]
You've got it. Ok to not show the panel on page refresh if they have already signed-up, or should it always show the thank-you message?

File: includes/classes/class-welcome-page.php [hunk 2]
```diff
@@ -205,4 +205,144 @@ public static function render_summary() {
...
+						<div class="opt-in-hide">
+							<h2 id="SIGNUP_HEADING">' . __( 'Get Notified of Upcoming Events', 'accessibility-checker' ) . '</h2>
+							<div>' . __( 'Join our email list and get event reminders and recordings in your inbox.', 'accessibility-checker' ) . '</div>
+							<div>
+								<div id="Zc_SignupSuccess" style="display: none;">' . __( 'Thank-you for signing up!', 'accessibility-checker' ) . '</div>
+							</div>
+							<form method="POST" id="zcampaignOptinForm" style="margin: 0px; width: 100%" action="https://zmp-glf.maillist-manage.com/weboptin.zc" target="_zcSignup">
+							
+							<div class="SIGNUP_FLD edac-mt-1" id="edac-opt-in-email">
+								<label style="font-weight: 600;" for="EMBED_FORM_EMAIL_LABEL">Email Address *</label>
```
Comment (SteveJonesDev) [inline includes/classes/class-welcome-page.php:243]: Accessibility: Change asterisk to '(Required)'

**Review by SteveJonesDev — CHANGES_REQUESTED** [review]
There are two hidden inputs at the bottom of the form that don't have form labels. I can't find them in the PHP so I'm assuming they are being added via JavaScript. I'm not sure if we can fix that or not.
Screenshot 2023-09-27 at 11 20 47 AM

Can we auto-populate the email input with the current logged-in user's email address?
If we do auto-populate we might want to inform screen readers.
Example:
```
<form>
  <label for="username">Username</label>
  <input type="text" id="username" aria-describedby="info-region" />
  <div id="info-region" aria-live="polite" class="info-message"></div>
</form>
```

```
document.addEventListener("DOMContentLoaded", function () {
  const usernameInput = document.getElementById('username');
  const infoRegion = document.getElementById('info-region');
  
  usernameInput.value = 'AutoPopulatedUser';
  infoRegion.textContent = 'Note: Username field has been auto-populated with "' + usernameInput.value + '".';
});
```

**Comment by boonedev** [comment]
The extra inputs are added by zoho's js onload. I could try to listen for them to exist then replace them with hidden fields if you think it's worth it?

Updates applied:
- no longer removes fields on submit, instead makes the email field readonly and disabled, and disables the submit button
- auto-populates WP email
- adds "Note: The Email Address field has been auto-populated with your WordPress email." notice & removes it when the email field is modified
- adds missing alt for spacer image

**Review by SteveJonesDev — CHANGES_REQUESTED** [review]
I think the a11y of this is worse. I think removing the form after submission is fine and showing the confirmation message. But we need to ensure the screen reader focuses on that message and reads it out on submit. Let's remove the aria-live message for now.

Here is an example of the confirmation message.
```
<!-- Form HTML -->
<form id="myForm">
  <!-- Form fields go here -->
  <input type="submit" value="Submit" />
</form>

<!-- Confirmation Message -->
<div id="confirmationMessage" tabindex="-1" role="alert" style="outline: none;">
  <!-- This will be populated with the confirmation message on form submit -->
</div>

<script>
// JavaScript to handle form submit
document.getElementById('myForm').addEventListener('submit', function(e){
  e.preventDefault(); // Prevent actual form submit
  
  // Populate the confirmation message
  document.getElementById('confirmationMessage').textContent = 'Your form has been submitted successfully.';
  
  // Set focus to the confirmation message
  document.getElementById('confirmationMessage').focus();
});
</script>
```

**Comment by boonedev** [comment]
So leave as-is and only remove `aria-describedby` and ` aria-live` attributes below:
```
<input name="CONTACT_EMAIL" id="EMBED_FORM_EMAIL_LABEL" aria-describedby="email-info-region">
<div class="edac-mt-3" id="email-info-region" aria-live="polite" class="info-message"></div>
```
Or re-add removing the form on submit and show a confirm message?

**Comment by SteveJonesDev** [comment]
This: Or re-add removing the form on submit and show a confirm message?

But only the form, not the whole subscribe container like it did before. Test it with Voice Over and ensure it reads out the confirmation message on submit.

If you need an example install Gravity Forms, create a test form, and set it to submit via Ajax.

**Comment by boonedev** [comment]
Updated.
- Adds confirmation message with role=alert 
- On submit, removes the form
 - On submit, populates confirmation message text and gives it focus
 - Tested w/Voice Over -on submit confirmation message is read-out

File: includes/classes/class-welcome-page.php [hunk 3]
```diff
@@ -205,4 +205,184 @@ public static function render_summary() {
...
+								<div id="Zc_SignupSuccess" style="display: none;"></div>
+							</div>
+							
+							<form method="POST" id="zcampaignOptinForm" style="margin: 0px; width: 100%" action="https://zmp-glf.maillist-manage.com/weboptin.zc" target="_zcSignup">
+							
+							<div class="SIGNUP_FLD edac-mt-3" id="edac-opt-in-email">
+								<label style="font-weight: 600;" for="EMBED_FORM_EMAIL_LABEL">Email Address (Required)</label>
+								<input style="max-width: 100%;" type="email" changeitem="SIGNUP_FORM_FIELD" name="CONTACT_EMAIL" id="EMBED_FORM_EMAIL_LABEL" aria-describedby="email-info-region" value="' . esc_attr( $email ) . '">
+								<div class="edac-mt-3" id="email-info-region" class="info-message">
+									' . esc_html( __( 'Note: The Email Address field has been auto-populated with your WordPress email.', 'accessibility-checker' ) ) . '
```
Comment (SteveJonesDev) [inline includes/classes/class-welcome-page.php:257]: Delete this message.

File: includes/classes/class-welcome-page.php [hunk 4]
```diff
@@ -205,4 +205,184 @@ public static function render_summary() {
...
+							<input type="hidden" id="viewFrom" value="URL_ACTION">
+							<input type="hidden" name="zx" id="cmpZuid" value="' . esc_attr( $zx ) . '">
+							<input type="hidden" id="zcld" name="zcld" value="' . esc_attr( $zcld ) . '">
+							<input type="hidden" id="zctd" name="zctd" value="' . esc_attr( $zctd ) . '">
+							<input type="hidden" id="zc_formIx" name="zc_formIx" value="' . esc_attr( $zc_form_ix ) . '">
+							<span style="display: none" id="dt_CONTACT_EMAIL">1,true,6,Contact Email,2</span>
+							</form>
+
+							<!-- Confirmation Message -->
+							<div id="confirmationMessage" tabindex="-1" role="alert" style="outline: none;">
```
Comment (SteveJonesDev) [inline includes/classes/class-welcome-page.php:288]: The confirmation message is being read out twice by the screen reader. Remove role="alert" and test that it's only being read out once.

**Review by SteveJonesDev — CHANGES_REQUESTED** [review]

**Comment by boonedev** [comment]
Removed role=alert and confirmed message is announced only once.

**Review by SteveJonesDev — APPROVED** [review]
Please merge.
